tokio-trace-proc-macros/src/lib.rs

```diff
@@ -33,11 +33,30 @@ pub fn trace(_args: TokenStream, item: TokenStream) -> TokenStream {
 
     let return_type = input.clone().decl.output;
     let params = input.clone().decl.inputs;
+    let param_names: Vec<Ident> = input
+        .clone()
+        .decl
+        .inputs
+        .into_iter()
+        .filter_map(|param| match param {
+            FnArg::Captured(ArgCaptured {
+                pat: Pat::Ident(PatIdent { ident, .. }),
```

Comment:
eventually we might want to handle destructuring patterns here, but this covers the common case...

Reply:
@hawkw Are there any examples you could link to (in this project or tokio) where we would want to destructure a pattern? I could look into extending this, but I don't think I have the use case in mind.